lib/instrumentation/trace-context.js

```diff
@@ -126,9 +132,12 @@ class TraceContext {
     })
   }
 
-  static startOrResume (traceparent, conf) {
-    return isValidHeader(traceparent)
-      ? resume(traceparent)
+  static startOrResume (childOf, conf) {
+    if (childOf instanceof TraceContext) return childOf.child()
+    if (childOf instanceof GenericSpan) return childOf._context.child()
```

Comment:
Could also just check it's an object and has a `_context` property that is `instanceof TraceContext`. That way you don't need the `GenericSpan` circular dependency an the complicated logic surrounding that.

Reply:
Fixed